src/proteus/dswx_hls.py

```diff
@@ -545,7 +583,8 @@ def _update_landcover_array(conglomerate_array, agg_sum, threshold,
 def create_landcover_mask(copernicus_landcover_file,
                           worldcover_file, output_file, scratch_dir,
                           mask_type, geotransform, projection, length, width,
-                          dswx_metadata_dict = {}, output_files_list = None):
+                          dswx_metadata_dict = {}, output_files_list = None,
```

Comment:
For Python, it is usually best to not use mutable objects (such as dicts or lists) as defaults, because the object will persist between calls to the function. Here's a short example to illustrate: https://dollardhingra.com/blog/python-mutable-default-arguments/ 

If the goal is to have the dictionary persist, could that intended behavior please be documented in the doc string? Otherwise, perhaps set `dswx_metadata_dict=None` and then use a `if dswx_metadata_dict=None: dswx_metadata_dict = {}` condition in the function's body.

Reply:
Very helpful, @nemo794 ! Thank you!